actors/user/src/main/java/org/sunbird/user/actors/UserStatusActor.java

```diff
@@ -0,0 +1,210 @@
+package org.sunbird.user.actors;
+
+import akka.actor.ActorRef;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.sunbird.actor.core.BaseActor;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.common.ElasticSearchUtil;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.ActorOperations;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.models.util.ProjectUtil;
+import org.sunbird.common.models.util.ProjectUtil.Status;
+import org.sunbird.common.models.util.PropertiesCache;
+import org.sunbird.common.request.ExecutionContext;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.learner.util.Util;
+import org.sunbird.models.user.User;
+import org.sunbird.services.sso.SSOManager;
+import org.sunbird.services.sso.SSOServiceFactory;
+import org.sunbird.telemetry.util.TelemetryUtil;
+import org.sunbird.user.dao.UserDao;
+import org.sunbird.user.dao.impl.UserDaoImpl;
+
+/**
+ * This actor will handle user status operation .
+ *
+ * @author Manzarul
+ * @author Amit Kumar
+ * @author sudhirgiri
+ */
+@ActorConfig(
+  tasks = {"unblockUser", "blockUser"},
+  asyncTasks = {}
+)
+public class UserStatusActor extends BaseActor {
+  private UserDao userDao = UserDaoImpl.getInstance();
+  private SSOManager ssoManager = SSOServiceFactory.getInstance();
+  private boolean isSSOEnabled =
+      Boolean.parseBoolean(PropertiesCache.getInstance().getProperty(JsonKey.IS_SSO_ENABLED));
+  private ActorRef systemSettingActorRef =
+      getActorRef(ActorOperations.GET_SYSTEM_SETTING.getValue());
+
+  /** Receives the actor message and perform the user status operation . */
+  @Override
+  public void onReceive(Request request) throws Throwable {
+    Util.initializeContext(request, JsonKey.USER);
+    ExecutionContext.setRequestId(request.getRequestId());
+    String operation = request.getOperation();
+
+    switch (operation) {
+      case "blockUser":
+        blockUser(request);
+        break;
+
+      case "unblockUser":
+        unblockUser(request);
+        break;
+
+      default:
+        onReceiveUnsupportedOperation("UserStatusActor");
+    }
+  }
+
+  /**
+   * Method to block the user , it performs only soft delete from Cassandra , ES , Keycloak
+   *
+   * @param actorMessage
+   */
+  private void blockUser(Request actorMessage) {
+
+    ProjectLogger.log("Method call  " + "deleteUser");
```

Comment:
The code for blockUser and unblockUser is very similar. Create a helper method updateUserStatus(Request request, boolean isBlocked). Why is the sync differently handled for block user and unblock user? Discuss with Amit and try to align this.